exporter: Add Wavefront exporter

mshvol opened the pull request:
Add Wavefront exporter to agent.

This exporter currently uses `exporterwrapper` on the existing opencensus-go's `trace.Exporter` work from:
https://github.com/wavefrontHQ/opencensus-exporter
The above exporter supports exporting both traces and stats.

Sample usage is provided in README as a PR hasn't been opened for opencensus-website yet.

I've left the copyright/license lines blank as I was unsure. Above repo is Apache-2.0.

CC @songy23 @odeke-em

mshvol commented:
The failed tests don't seem to be related to this code (Some prometheus and Thrift tests are failing). Please tell me if anything is wrong from my side and I'll fix it.

songy23 commented:
@mshvol Thanks for the contribution! 

> I've left the copyright/license lines blank as I was unsure. Above repo is Apache-2.0.

Unfortunately AFAIK we need the source code files to be under the OpenCensus license. Apache-2.0 is fine for us since this repo is also under Apache-2.0 (https://github.com/census-instrumentation/opencensus-service/blob/master/LICENSE). /cc @jparsana kindly correct me if I'm wrong.

> The failed tests don't seem to be related to this code (Some prometheus and Thrift tests are failing). Please tell me if anything is wrong from my side and I'll fix it.

You're right it's because of the upgraded OC-Go version. We're actively working on resolving the dependency issue. Once we updated the OC-Go version in this repo this should be fixed. No additional action required from your side :)

songy23 commented on exporter/wavefrontexporter/wavefront_test.go:
```diff
@@ -0,0 +1,3 @@
+package wavefrontexporter
+
+// TODO: Add tests.
```
If there're no tests at this moment then you need to put a `nocover` in this package, similar to https://github.com/census-instrumentation/opencensus-service/blob/master/internal/collector/jaeger/.nocover.

mshvol replied:
From `check-test-files.sh`, `nocover` seems to work only when there are no `*_test.go` files.
I've still added one. But, the following exporters also have dummy test files without `nocover`.

```
./jaegerexporter/jaegerexporter_test.go
./kafkaexporter/kafkaexporter_test.go
./stackdriverexporter/stackdriverexporter_test.go
./datadogexporter/datadog_test.go
./honeycombexporter/honeycombexporter_test.go
./exporterwrapper/exporterwrapper_test.go
```

mshvol commented:
Mirrored the license headers found in other exporters.

mshvol commented:
> Hi @mshvol as mentioned by @songy23 we need to put this on hold until the oc-go changes are merged.
> 
@pjanotti When do you plan on upgrading to the newer oc-go? Do you have a timeline for this?

pjanotti commented:
@mshvol we need to wait for https://github.com/census-instrumentation/opencensus-go/pull/1118 and updated release of same project.

songy23 commented:
Hi @mshvol, opencensus versions have been updated, please try rebase and do a force push.

mshvol commented:
Sorry for the delay. I've rebased to latest commit and incorporated @pjanotti's suggestions. The [oc-go exporter](https://github.com/wavefrontHQ/opencensus-exporter) has also been updated to use oc-go v0.21.0.
